File: Aquila/aquila.cpp
#include <stdlib.h>

#include "aquila.h"

namespace aquila {
	const char* version() {
		return __DRAGO_AQUILA;
	}

	void add_node(int id) {

	}

	void remove_node(int id) {

	}

	void connect_nodes(int a, int b) {

	}

	void disconnect_nodes(int a, int b) {

	}

	void clear_nodes() {

	}

	void navigate() {

	}

	Node::Node(Graph* graph) {
	}

	Node::~Node() {
	}

	std::map<int, Node>* Node::GetConnections() {
		return nullptr;
	}

	void Node::Connect(Node* node) {
	}

	void Node::Disconnect(Node* node) {
	}

	Graph::Graph() {
	}

	Graph::~Graph() {
	}

	int Graph::AddNode() {
		return 0;
	}

	void Graph::RemoveNode(Node* node) {
	}

	void Graph::ConnectNodes(Node* a, Node* b) {
	}

	void Graph::DisconnectNodes(Node* a, Node* b) {
	}

	void Graph::ClearNodes() {
	}

	void Graph::Navigate(Node* source, Node* destination) {
	}
}

File: Aquila/Aquila.cs
using System;
using System.Collections.Generic;

namespace Aquila {
    public class Aquila {
        static int id_record = 0;

        public int ID { get; private set; }

        private Dictionary<int, Node> nodes = new Dictionary<int, Node>();

        public Aquila() {
            ID = id_record++;
        }

        public int AddNode(double data) {
            Node addition = new Node(data);
            nodes[addition.ID] = addition;
            return addition.ID;
        }

        public void RemoveNode(int id) {
            if (nodes.ContainsKey(id)) {
                nodes.Remove(id);
            }
        }
    }

    public class Node {
        static int id_record = 0;

        public int ID { get; private set; }
        public double Data { get; set; }

        private HashSet<Node> connections = new HashSet<Node>();

        public Node(double data) {
            ID = id_record++;
            Data = data;
        }

        public void Connect(Node other) {
            if (!connections.Contains(other)) {
                connections.Add(other);
                other.connections.Add(this);
            }
        }

        public void Disconnect(Node other) {
            connections.Remove(other);
            other.connections.Remove(this);
        }
    }
}
